site/app/templates/polls/ViewPoll.twig

```diff
@@ -95,6 +107,50 @@
 </div>
 
 <script>
+
+    {% if timer_enabled %}
+
+        var endDate = new Date('{{ end_time }}');
+        var timerId;
+        var ended = false;
+
+        function updateTimer() {
+            var now = new Date();
+            var timeRemaining = endDate - now;
+            var timerDisplayElement = document.getElementById('timerDisplay');
+            var timerElement = document.getElementById('timer');
+
+            // Show the timer element once we're ready to start updating it
+            timerElement.style.display = 'block';
+
+            if (timeRemaining <= 0) {
+                if (timerElement) {
+                    timerElement.innerHTML = "Poll Ended";
```

Comment:
Using `innerHTML` poses a security risk because user input can be parsed as HTML, leading to potential HTML injection and harmful script execution. Use `textContent` or `text()` instead of `html()` or `innerHTML` to prevent this.

Reply:
Does this apply if the `innerHTML` is not taking user input?

Reply:
Yes, it is good practice to avoid innerHTML whenever possible, even when content is not user-generated. Better future proof, and consistency across the codebase.

Reply:
done in [70785f6](https://github.com/Submitty/Submitty/pull/10337/commits/70785f627ffdd44e5beed80aa95e15de48a82be3)